### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'file_sums'

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#4953]
after upgrade to conan 1.14, the following failure happen randomly, but frequently:

```
 >> OK! '[secure]' user logged in 'upload_repo' 
Uploading to remote 'upload_repo':
Uploading libcurl/7.64.1@bincrafters/stable to remote 'upload_repo'
Compressing recipe sources...                                         [==================================================]
Traceback (most recent call last):
  File "/opt/pyenv/versions/3.7.1/bin/run_create_in_docker", line 11, in <module>
    load_entry_point('conan-package-tools==0.25.1', 'console_scripts', 'run_create_in_docker')()
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/cpt/run_in_docker.py", line 50, in run
    runner.run()
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/cpt/runner.py", line 130, in run
    self._upload, package_id)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/cpt/uploader.py", line 16, in upload_packages
    self._upload_artifacts(reference, upload, package_id)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/cpt/uploader.py", line 50, in _upload_artifacts
    retry=int(self._upload_retry))
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/client/conan_api.py", line 93, in wrapper
    return f(*args, **kwargs)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/client/conan_api.py", line 868, in upload
    retry_wait, integrity_check, policy, remote_name, query=query)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/client/cmd/uploader.py", line 88, in upload
    integrity_check, policy, remote, upload_recorder)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/client/cmd/uploader.py", line 190, in _upload_ref
    self._upload_recipe(ref, conanfile, retry, retry_wait, policy, recipe_remote)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/client/cmd/uploader.py", line 230, in _upload_recipe
    remote, remote_manifest)
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/client/cmd/uploader.py", line 344, in _recipe_files_to_upload
    if remote_manifest == local_manifest:
  File "/opt/pyenv/versions/3.7.1/lib/python3.7/site-packages/conans/model/manifest.py", line 132, in __eq__
    return self.file_sums == other.file_sums
AttributeError: 'NoneType' object has no attribute 'file_sums'
Traceback (most recent call last):
  File "build.py", line 51, in <module>
    builder.run()
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/cpt/packager.py", line 473, in run
    self.run_builds(base_profile_name=base_profile_name)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/cpt/packager.py", line 586, in run_builds
    docker_entry_script=self.docker_entry_script)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/cpt/runner.py", line 266, in run
    raise Exception("Error building: %s" % command)
Exception: Error building: sudo -E docker run --rm -v "/home/travis/build/bincrafters/conan-libcurl:/home/conan/project" -e CONAN_GCC_VERSIONS="6" -e CONAN_DOCKER_IMAGE="conanio/gcc6" -e CONAN_LOGIN_USERNAME="[secure]" -e CONAN_PASSWORD="[secure]" -e CONAN_CHANNEL="stable" -e CONAN_UPLOAD="https://api.bintray.com/conan/bincrafters/public-conan@True@upload_repo" -e CONAN_REMOTES="https://api.bintray.com/conan/bincrafters/public-conan@True@remote0" -e CONAN_REFERENCE="libcurl/7.64.1@bincrafters/stable" -e CPT_PROFILE="@@include(default)@@@@[settings]@@arch=x86@@build_type=Release@@compiler=gcc@@compiler.version=6@@[options]@@libcurl:shared=True@@[env]@@@@[build_requires]@@@@" -e CONAN_USERNAME="bincrafters" -e CONAN_TEMP_TEST_FOLDER="1" -e CPT_UPLOAD_ENABLED="True" -e CPT_UPLOAD_RETRY="3" -e CPT_CONANFILE="conanfile.py" -e PIP_PROGRESS_BAR="off" -e PIP_DISABLE_PIP_VERSION_CHECK="1"   conanio/gcc6 /bin/sh -c " cd project &&  run_create_in_docker "
```
logs: https://travis-ci.com/bincrafters/conan-libcurl/jobs/192040098

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by SSE4** [comment]
failing code:
```
https://github.com/conan-io/conan/blame/e3c6463c1078d54712acb97c0c60822b563169b5/conans/model/manifest.py#L132
```
the potential fix is to rewrite into something like that:
```
def __eq__(self, other):
        """ Two manifests are equal if file_sums
        """
        if self is None and other is None:
            return True
        elif self is None or other is None:
            return False
        return self.file_sums == other.file_sums
```
as `other` object could be None in some circumstances, it obviously has to be checked here

**Comment by SSE4** [comment]
or it might be better to add checks here, I am not really sure:
https://github.com/conan-io/conan/blob/9d6fb5b026db594bfc05784c8f1cf1f8f200704e/conans/client/cmd/uploader.py#L346
https://github.com/conan-io/conan/blob/9d6fb5b026db594bfc05784c8f1cf1f8f200704e/conans/client/cmd/uploader.py#L367

**Comment by SSE4** [comment]
seems to be related to the https://github.com/conan-io/conan/issues/4698

**Comment by jgsogo** [comment]
When you know what is happening, let us know if it is related to a bug in Bintray, if there is something we can handle on our side... thanks for having a look into it.

**Comment by lasote** [comment]
Any news about this? @SSE4 the same issue has been reported a couple of times and I suffered it myself.  Did you manage to know why it is random?

**Comment by lasote** [comment]
I think it could be a concurrency issue, the client checks the remote manifest and it is none, then it checks the snapshot and is != None, because maybe another job is uploading the files at that moment or between the two checks. I would open a PR I don't know if solving it but at least to increasing the information about the error.

**Comment by SSE4** [comment]
I've been trying to reproduce it in various ways.
I was running parallel conan upload jobs, and script removing various files from bintray in addition to them, but never got this error locally, unfortunately. so, I am pretty confident that removal of manifests or another data on bintray doesn't render in this error. the only thing I was getting are bunch of random 405 errors on upload.
I think we need to study code deeper in order to understand in which consequences it may appear.

**Comment by memsharded** [comment]
Closed by #5014, will be released in 1.14.4
